runtime/queries/column_timeseries.go

```diff
@@ -131,19 +139,37 @@ func (q *ColumnTimeseries) Resolve(ctx context.Context, rt *runtime.Runtime, ins
 		}()
 
 		rows, err := olap.Execute(ctx, &drivers.Statement{
-			Query:    `SELECT * from "` + temporaryTableName + `"`,
+			Query:    fmt.Sprintf(`SELECT * FROM %s`, temporaryTableName),
 			Priority: priority,
 		})
 		if err != nil {
 			return err
 		}
 
-		results, err := convertRowsToTimeSeriesValues(rows, len(measures)+1, tsAlias)
-		rows.Close()
+		structs, err := rowsToData(rows)
 		if err != nil {
 			return err
 		}
 
+		results := make([]*runtimev1.TimeSeriesValue, 0, len(structs))
+		for _, s := range structs {
+			t, err := time.Parse(time.RFC3339, s.Fields[tsAlias].GetStringValue())
+			if err != nil {
+				return err
+			}
+
+			results = append(results, &runtimev1.TimeSeriesValue{
+				Ts:      timestamppb.New(t),
+				Bin:     s.Fields["bin"].GetNumberValue(),
+				Records: s,
+			})
+			delete(s.Fields, tsAlias)
+			delete(s.Fields, "bin")
+		}
```

Comment:
1. Looks like we could save a bunch of allocs/serialization/deserialization by merging this with `rowsToData`? So the `TimeSeriesValue` is created directly when iterating over `rows`. Seems worth it for performance
2. The `s.Fields["bin"]` is not available for this SQL query – mistake? It also might clash with a measure name here

Reply:
2. Yes, it's only used for Spark query.

Reply:
1. Yes, I had code flexibility in mind, right now if the SQL query is changed and we have other types then we need less code changes as `pbutil.ToStruct` will handle conversions, otherwise if it's rewritten without `ToStruct` then one needs to add conversion for new types. But yes, I can optimise it more.